pkg/steps/project_image.go

```diff
@@ -83,10 +64,16 @@ func (s *projectDirectoryImageBuildStep) run(ctx context.Context) error {
 				Name: source,
 			},
 			Paths: []buildapi.ImageSourcePath{{
-				SourcePath:     fmt.Sprintf("%s/%s/.", workingDir, s.config.ContextDir),
+				SourcePath:     fmt.Sprintf("%s/.", path.Join(baseDir, contextDir)),
 				DestinationDir: ".",
 			}},
 		})
+	} else {
+		sourceTag = s.config.From
+	}
```

Comment:
I *think* this would be slightly more readable in the following form:

```go
var sourceTag api.PipelineImageStreamTagReference = s.config.From
// Three options to set `sourceTag` to somethign different
...
if sourceTag != s.config.From { 
  // if we are changing the source, we need to make sure it's mounted in
}
```